usaspending_api/etl/swap_in_new_table_strategies.py

```diff
@@ -0,0 +1,944 @@
+
+            # If `delta_table_load_version_key` exists in the table_spec, this table could potentially be loaded incrementally,
+            # therefore we need to keep track of the last version of the table persisted to the live table. In this case, that
+            # would be the last version swapped from temp to live. The `last_version_to_staging` represents the last version
+            # either that was either fully copied to the temp table, or the last version incrementally copied to staging tables,
+            # so here we can use it as the new version copied to live.
+            table_spec = TABLE_SPEC.get(self._options["table"])
+            delta_table_load_version_key = (
+                table_spec["delta_table_load_version_key"] if table_spec is not None else None
+            )
+
+            if delta_table_load_version_key is not None:
+                last_version_to_staging, last_version_to_live = get_last_delta_table_load_versions(
+                    delta_table_load_version_key
+                )
+                update_last_live_load_version(delta_table_load_version_key, last_version_to_staging)
+
+    def _validate_options(self) -> None:
+        """Validates the combination of options provided to this command. Will
+        raise an exception if any are invalid
+        """
+        is_undo = self._options["undo"]
+        table = self._options["table"]
+        load_step = LoadTrackerStepEnum(self._load_tracker.load_tracker_step.name)
+        latest_load_tracker_load_type = LoadTracker.get_latest_load_tracker_type(load_step.value)
+        if latest_load_tracker_load_type == LoadTrackerLoadTypeEnum.INCREMENTAL_LOAD and is_undo:
+            raise ArgumentError(
+                f"The load tracker table indicates the last load type for {table} was an {LoadTrackerLoadTypeEnum.INCREMENTAL_LOAD.value}. You cannot use the option undo when the last load was an {LoadTrackerLoadTypeEnum.INCREMENTAL_LOAD.value}"
+            )
+
+    def _create_new_views(self, cursor):
+        for dep_view in self.dep_views:
+            # Note: Despite pointing at the new temp table at first,
+            #       the recreated views will automatically update and stay pointed to it in the swap via postgres
+            self.logger.info(f"Recreating dependent view: {dep_view['dep_view_fullname']}{self.source_suffix}")
+            dep_view_sql = dep_view["dep_view_sql"].replace(self.curr_table_name, self.temp_table_name)
+            mv_s = "MATERIALIZED " if dep_view["is_matview"] else ""
+            cursor.execute(
+                f"DROP {mv_s}VIEW IF EXISTS {self.temp_schema_name}.{dep_view['dep_view_name']}{self.source_suffix};"
+            )
+            cursor.execute(
+                f"CREATE {mv_s}VIEW {self.temp_schema_name}.{dep_view['dep_view_name']}{self.source_suffix} "
+                f"AS ({dep_view_sql});"
+            )
+
+    def _dependent_views(self, cursor, curr_schema_name, curr_table_name):
```

Comment:
I believe methods like this that are implemented in both strategies should be moved into the base class

Reply:
Be warned moving this to a shared class couples these classes together. I made the change you requested